### PR TITLE
Zombienet: BFT threshold dispute test

### PR DESCRIPTION
**Pull request by sandreim**
Related to https://github.com/paritytech/polkadot/issues/4776

### CODE REVIEW

File: zombienet_tests/functional/0003-parachains-bft-disputes.feature [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+honest-validator-6: reports node_roles is 4
+honest-validator-7: reports node_roles is 4
+honest-validator-8: reports node_roles is 4
+honest-validator-9: reports node_roles is 4
+malus-validator-0: reports node_roles is 4
+malus-validator-1: reports node_roles is 4
+malus-validator-2: reports node_roles is 4
+
+
+honest-validator-0: parachain 2000 block height is at least 4 within 180 seconds
```
Comment (drahnr) [inline zombienet_tests/functional/0003-parachains-bft-disputes.feature:35]: 4 blocks after 180 is a safeguard for cascaded startup delay, correct?

File: zombienet_tests/functional/0003-parachains-bft-disputes.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+honest-validator-0: reports parachain_disputes_finality_lag is lower than 4
+honest-validator-1: reports parachain_disputes_finality_lag is lower than 4
+honest-validator-2: reports parachain_disputes_finality_lag is lower than 4
+honest-validator-3: reports parachain_disputes_finality_lag is lower than 4
+honest-validator-6: reports parachain_disputes_finality_lag is lower than 4
+honest-validator-7: reports parachain_disputes_finality_lag is lower than 4
+honest-validator-8: reports parachain_disputes_finality_lag is lower than 4
+honest-validator-9: reports parachain_disputes_finality_lag is lower than 4
+
+honest-validator-0: reports parachain_candidate_disputes_total is at least 4 within 15 seconds
```
Comment (drahnr) [inline zombienet_tests/functional/0003-parachains-bft-disputes.feature:52]: We expect all 4 blocks produced to be disputed. :heavy_check_mark:

File: zombienet_tests/functional/0003-parachains-bft-disputes.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+malus-validator-1: reports node_roles is 4
+malus-validator-2: reports node_roles is 4
+
+
+honest-validator-0: parachain 2000 block height is at least 4 within 180 seconds
+honest-validator-1: parachain 2001 block height is at least 4 within 180 seconds
+honest-validator-2: parachain 2002 block height is at least 4 within 180 seconds
+honest-validator-3: parachain 2003 block height is at least 4 within 180 seconds
+
+honest-validator-3: log line contains "reverted due to a bad parachain block" within 180 seconds
```
Comment (drahnr) [inline zombienet_tests/functional/0003-parachains-bft-disputes.feature:40]: That should be the case for all honest validators, correct?

Comment (sandreim) [reply zombienet_tests/functional/0003-parachains-bft-disputes.feature:40]: Yes

File: zombienet_tests/functional/0003-parachains-bft-disputes.feature (hunk 1 above)
Comment (sandreim) [reply zombienet_tests/functional/0003-parachains-bft-disputes.feature:35]: yup

**Pull request closed without merging**